app/scripts/modules/core/src/artifact/ArtifactIconService.ts

```diff
@@ -0,0 +1,24 @@
+const kubernetesLogoPath = require('./icons/kubernetes-artifact.svg');
+const unknownArtifactPath = require('./icons/unknown-type-artifact.svg');
+
+export class ArtifactIconService {
+  private static typeMap: Map<string, string> = new Map();
+
+  public static registerType(type: string, iconPath: string) {
+    this.typeMap.set(type, iconPath);
+  }
+
+  public static getPath(type: string) {
+    return this.typeMap.get(type) || unknownArtifactPath;
+  }
+}
+
+ArtifactIconService.registerType('docker/image', require('./icons/docker-image-artifact.svg'));
+ArtifactIconService.registerType('kubernetes/manifest', kubernetesLogoPath);
```

Comment:
I think this should be able to cover `kubernetes/*`, since technically any kind should be supported